src/persist-types/src/dyn_struct.rs

```diff
@@ -218,17 +218,25 @@ impl DynStructCol {
         let (mut fields, mut arrays) = (Vec::new(), Vec::new());
         for (name, _stats_fn, col) in self.cols() {
             let (array, is_nullable) = col.to_arrow();
+            if array.len() == 0 && self.len != 0 {
```

Comment:
Can we stick a comment here? (Also this seems slightly wrong/odd/surprising, but maybe the comment will enlighten me)

Reply:
Shoot, sorry I left this in, it was a remnant of when we used to map a `StructArray` with no fields to an empty `NullArray`. Because we changed that to be a NullArray with a length, we no longer need this check. Removed!